src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs

```diff
@@ -90,7 +90,7 @@ private void CreateModels(IDictionary<InputModelType, ModelTypeProvider> models,
         {
             //only want to create one instance of the default derived per polymorphic set
             ModelTypeProvider? defaultDerivedType = null;
-            bool isBasePolyType = derivedTypesArray.Count > 0 && model.DiscriminatorPropertyName is not null;
+            bool isBasePolyType = model.DiscriminatorPropertyName is not null;
```

Comment:
Even it has no child, it is still the base type of a polymorphism

Reply:
the condition is changed to what you mean